osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -286,11 +286,10 @@ public WorkingBeatmap GetWorkingBeatmap(BeatmapInfo beatmapInfo, WorkingBeatmap
         {
             if (beatmapInfo?.ID > 0 && previous != null && previous.BeatmapInfo?.ID == beatmapInfo.ID)
                 return previous;
-
             if (beatmapInfo?.BeatmapSet == null || beatmapInfo == DefaultBeatmap?.BeatmapInfo)
                 return DefaultBeatmap;
 
-            if (beatmapInfo.BeatmapSet.Files == null)
+            if (beatmapInfo.BeatmapSet.Files == null || beatmapInfo.ID == 0)
```

Comment:
I don't think I follow this either. What purpose would requerying a beatmap of an ID of 0 have, if the requery is performed using the exact same ID? I guess I could go and find out, but I figure I might as well ask directly...

Reply:
Autoincrement in basically every relational database starts from `1`. Maybe this deserves an inline comment?

The original intention of checking for `Files==null` looks to be exactly this case, but some tests look to populate `Files` with an empty list, which is what I was running in to.

Reply:
I am aware that autoincrement/identity usually starts at 1. Which is exactly where my confusion comes from, as there should never be a matching beatmap with ID 0, so I can't see how a requery for that ID can ever succeed.

Additionally I don't really get what an empty `Files` list has to do with this check.

Reply:
Aha right, you have a point. I didn't realise the following lookup was based on that same `ID`. Will look at a better way of detailing this case.